back/app/models/project.rb

```diff
@@ -263,6 +263,7 @@ def remove_old_folder_moderators
     old_folder_moderators.each do |moderator|
       next unless moderator.moderatable_project_ids.include?(id)
 
+      project.update(default_assignee_id: nil) if default_assignee_id == moderator.id
```

Comment:
@sebastienhoorens The problem was that we were removing the project moderator roles here (when moving the project out of a folder, but not removing the project.default_assignee if any of the project moderators were that relation).

Thus, the check of `project.valid?` in the `check_publication_inconsistencies!` would catch this, via the `assignee_can_moderate_project` validation in the model patch [here](https://github.com/CitizenLabDotCo/citizenlab/blob/b5a94f3bb98e62b2330f378e5a2e27fa04a6d5e2/back/engines/commercial/idea_assignment/app/models/idea_assignment/extensions/project.rb#L13-L21).

I could put this in a separate after_save hook (e.g. `after_save: remove_default_assignee_if_cannot_moderate`), but that would mean another check to see if the `default_assignee` can moderate the project, which seems unnecessary, given that at this point we know they cannot moderate as we are removing this role. WDYT?